### PR TITLE
Fix the issue of javadocJar not including documentation & Publish sourcesJar to the GitHub Release page.

### PR DESCRIPTION
**Pull request by overcat**
* Fix the issue of javadocJar not including documentation.
* Publish sourcesJar to the GitHub Release page.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -97,6 +96,17 @@ tasks {
...
 
+    val javadocJar by creating(Jar::class) {
+        manifest {
+            attributes["Implementation-Title"] = "stellar-sdk"
+            attributes["Implementation-Version"] = version
+        }
+        archiveClassifier = "javadoc"
+        archiveFileName = "stellar-sdk-javadoc.jar"
+        dependsOn(javadoc)
+        from(javadoc.get().destinationDir) // It needs to be placed after the javadoc task, otherwise it cannot read the path we set.
```
Comment (sreuland) [inline build.gradle.kts:107]: subtle, thanks for including docs to explain this.

**Review by sreuland — APPROVED** [review]

**Comment by overcat** [comment]
When I retriggered CI, Circle CI was still here, let me try to close this PR and create a new one.

**Comment by overcat** [comment]
HI @sreuland, please check my Discord message. Due to incorrect project configuration settings, we are unable to merge this PR at the moment.

**Comment by sreuland** [comment]
@overcat , we should be able to resolve this pr action check by tomorrow, will let you know when it's effective.

**Comment by sreuland** [comment]
@overcat , the action checks have been updated, 'android-test` needs to run still and then pr should allow merge

**Comment by overcat** [comment]
@sreuland, it seems that the settings are still incorrect, `android-test` should be changed to `android-test (api ${{ matrix.emulator.api-level }})`

**Comment by sreuland** [comment]
@overcat , I have submitted a change to only do pr action checks on build and test jobs, should land shortly.

**Comment by overcat** [comment]
@sreuland, I found [the discussion](https://github.com/orgs/community/discussions/12377) initiated by leighmcculloch. If we want to add Android checks, we should add `android-test (api 23)` and `android-test (api 32)`.

**Comment by sreuland** [comment]
@overcat , based on test values - https://github.com/overcat/java-stellar-sdk/blob/master/.github/workflows/test-deploy.yml#L78-L79

that would be `android-test (api 26) and android-test (api 32)` correct? I'm not sure if the `target` test dimension needs to be included as well, but can try this, or we just leave it to `build and test` for now to get pr's moving and try this later?

**Comment by sreuland** [comment]
@overcat, pr checks are resolved for now, based on build/test jobs.

**Comment by overcat** [comment]
> @overcat , based on test values - https://github.com/overcat/java-stellar-sdk/blob/master/.github/workflows/test-deploy.yml#L78-L79
> 
> that would be `android-test (api 26) and android-test (api 32)` correct? I'm not sure if the `target` test dimension needs to be included as well, but can try this, or we just leave it to `build and test` for now to get pr's moving and try this later?

@sreuland, the reason why I hope it is `android-test (api 23)` instead of `android-test (api 26)` is because I set the `target` to **23** in #543, and I want to avoid modifying the GitHub settings again later. But now you have already set it as `android-test (api 26)`, so I think we have to make another modification in the settings and set it as `android-test (api 23)`.

**Comment by sreuland** [comment]
ok, let's wait for remaining work to converge back to master, and then we can try adding `android-test (api <ver>)` to the pr action checks, for now, as mentioned `android test` jobs are not required, just `build/test`

**Pull request merged**